### PR TITLE
enhance assertions in test/e2e/windows

### PR DESCRIPTION
**Pull request by hunknownz**
Signed-off-by: hunknownz <klercc37@gmail.com>

#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Better output for developers when tests fail.

#### Which issue(s) this PR fixes:

Fixes part of https://github.com/kubernetes/kubernetes/issues/105678

#### Special notes for your reviewer:

After this PR:
```
.../kubernetes/test/e2e/windows$ git grep 'Expect.*Equal([^,]*, *\(true\|false\)' | wc -l
0
```

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/windows/host_process.go [hunk 1]
```diff
@@ -568,10 +569,10 @@ var _ = SIGDescribe("[Feature:WindowsHostProcessContainers] [MinimumKubeletVersi
...
 		// This done so the test can be run in parallel with other tests which may start HostProcess containers on the same node.
 		ginkgo.By("Ensuring metrics were updated")
-		framework.ExpectEqual(beforeMetrics.StartedContainersCount < afterMetrics.StartedContainersCount, true, "Count of started HostProcess containers should increase")
-		framework.ExpectEqual(beforeMetrics.StartedContainersErrorCount < afterMetrics.StartedContainersErrorCount, true, "Count of started HostProcess errors containers should increase")
-		framework.ExpectEqual(beforeMetrics.StartedInitContainersCount < afterMetrics.StartedInitContainersCount, true, "Count of started HostProcess init containers should increase")
-		framework.ExpectEqual(beforeMetrics.StartedInitContainersErrorCount < afterMetrics.StartedInitContainersErrorCount, true, "Count of started HostProcess errors init containers should increase")
+		gomega.Expect(beforeMetrics.StartedContainersCount).To(gomega.BeNumerically("<", afterMetrics.StartedContainersCount), fmt.Sprintf("Count of started HostProcess containers should increase: >= %v", afterMetrics.StartedContainersCount))
```
Comment (pohly) [inline test/e2e/windows/host_process.go:572]: Here and below the gomega.BeNumerically already takes care of printing the two values, so they don't need to be included in the explanation.

Also note that Expect does the printf formatting itself, so fmt.Sprintf would not be needed here.

**Review by pohly — APPROVED** [review]
/lgtm

**Comment by hunknownz** [comment]
@pohly which owner should I assign?
Screen Shot 2022-06-02 at 9 21 38 PM

**Comment by marosset** [comment]
I can take a look today.

/test pull-kubernetes-e2e-capz-windows-containerd

**Comment by marosset** [comment]
The tests that did fail are known issues are should be addressed with https://github.com/kubernetes/kubernetes/pull/110235

**Comment by hunknownz** [comment]
> I can take a look today.
> 
> /test pull-kubernetes-e2e-capz-windows-containerd

Thanks

**Pull request merged**